src/components/top-bar.tsx

```diff
@@ -15,70 +15,119 @@ export class TopBarComponent extends BaseComponent<IProps, IState> {
+          {this.renderCoilButton(leftMagType, "left", primaryDisabledClass, this.handleClickLeftMagnetCoilButton)}
+          <svg className="divider">
+            <use xlinkHref="#icon-divider"/>
+          </svg>
+          <div className={"label add right " + rightDisabledClass}>Add a 2nd magnet (movable)</div>
+          {this.renderBarButton(rightMagType, "right", rightDisabledClass, this.handleClickRightMagnetBarButton)}
+          {this.renderCoilButton(rightMagType, "right", rightDisabledClass, this.handleClickRightMagnetCoilButton)}
+          {this.renderRemoveButton(removeDisabledClass)}
         </div>
-        {this.renderMagnetButton(secondaryMagText, secondaryMagType, "right")}
+        {this.renderMagnetButton(leftMagText, leftMagType, "left")}
+        {this.renderMagnetButton(rightMagText, rightMagType, "right")}
       </div>
     );
   }
 
   private renderMagnetButton = (text: string, magType: MagnetType | null, posClass: string) => {
+    const selectClass = magType ? " " : "selectable ";
+    return (
+      <div className={"button " + selectClass} onClick={this.handleClickMagnetButton}>
+        <svg className={"icon nav-back2 " + posClass}>
+          <use xlinkHref={"#icon-" + posClass + "-nav-back2"} />
+        </svg>
+        <svg className={"icon nav-back1 " + posClass}>
+          <use xlinkHref={"#icon-" + posClass + "-nav-back1"} />
+        </svg>
+        <svg className={"icon nav-select-back " + selectClass + posClass}>
+          <use xlinkHref={"#icon-" + posClass + "-nav-select-back"} />
+        </svg>
+        <svg className={"icon nav-select-outline " + posClass}>
+          <use xlinkHref={"#icon-" + posClass + "-nav-select-outline"} />
+        </svg>
+        {magType
+        ? <svg className={"icon nav-magnet " + posClass + " " + magType}>
+            <use xlinkHref={"#icon-" + posClass + "-nav-" + magType + "-magnet"} />
+          </svg>
+        : <div className={"title " + posClass}>{text}</div>}
+      </div>
+    );
+  }
+
+  private renderBarButton = (magType: MagnetType | null, posClass: string,
+                             disabledClass: string, onClickFunction: () => void) => {
+    const selectClass = !magType || magType !== "bar" ? "selectable " : " ";
     return (
-      <div className={"magnet-button " + posClass} onClick={this.handleClickMagnetButton}>
-      {magType
-        ? <img src={"assets/magnet-" + magType + "-icon.png"} className={"icon " + posClass}/>
-        : <div className={"unselected " + posClass}>{text}</div>}
+      <div className={"button " + selectClass + disabledClass} onClick={onClickFunction}>
+        <svg className={"icon top-magnet-back bar " + selectClass + posClass}>
+          <use xlinkHref={"#icon-top-bar-magnet-back-" + posClass} />
+        </svg>
+        <svg className={"icon top-magnet-outline bar " + posClass}>
+          <use xlinkHref={"#icon-top-bar-magnet-outline-" + posClass} />
+        </svg>
+        {!magType || magType !== "bar"
```

Comment:
Just checking `magType !== "bar"` should be sufficient here

Reply:
Yup, probably a remnant from changes to handle showing the bar icon when we have either no mag or a coil mag.